src/Hash.mo

```diff
@@ -6,6 +6,9 @@ module Hash {
         reset() : ();
         // Returns the number of bytes that sum will return.
         size() : Nat;
+        // Return the hash.
+        // TODO: Should probably rename this
+        checkSum() : [Nat8];
```

Comment:
I added it to avoid repeating myself in the SHA3_* by reusing `New`. The `checkSum` function in SHA2.mo seemed to be the main part of the hashing too. Thought it'd be good to add. I can define the parameters like `H256` instead though.